baseplate.go

```diff
@@ -77,17 +77,42 @@ type Server interface {
 	Serve() error
 }
 
-// Serve runs the given Server until it is given an external shutdown signal
-// using runtimebp.HandleShutdown to handle the signal and shut down the
-// server gracefully.  Returns the (possibly nil) error returned by "Close" or
+// ServeArgs provides a list of arguments to Serve.
+//    server: Mandatory. Indicates the Server that should be run until
+//            receiving a shutdown signal.
+//    preShutdown: Optional. Indicates a slice of io.Closers that should
+//						be gracefully shut down before the server upon receipt of a
+//            shutdown signal.
+//    preShutdown: Optional. Indicates a slice of io.Closers that should
+//						be gracefully shut down after the server upon receipt of a
+//            shutdown signal.
+type ServeArgs struct {
+	server       Server
+	preShutdown  []io.Closer
+	postShutdown []io.Closer
```

Comment:
2 things:

1. You actually want to document the fields just above the field itself, not when documenting the struct itself.
2. These need to be capitalized so they are exported, as it is, only things in this package actually have access to these fields (so they could never actually be set by a service).

Reply:
Ooh, I didn't realize that capitalization exporting applied to fields of structs as well as the struct itself. Clever!